packages/arb-token-bridge-ui/src/components/TokenModal/TokenModalUtils.ts

```diff
@@ -0,0 +1,99 @@
+import { ERC20BridgeToken, TokenType } from 'token-bridge-sdk'
+
+import { TokenListWithId } from '../../tokenLists'
+
+export interface SearchableToken extends ERC20BridgeToken {
+  tokenLists: number[]
+}
+
+export type SearchableTokenStorage = { [address: string]: SearchableToken }
+
+export function tokenListsToSearchableTokenStorage(
+  tokenLists: TokenListWithId[],
+  l1ChainId: string,
+  l2ChainId: string
+): SearchableTokenStorage {
+  return (
+    tokenLists
+      //
+      .reduce((acc: SearchableTokenStorage, tokenList: TokenListWithId) => {
+        tokenList.tokens.forEach(token => {
+          const address = token.address.toLowerCase()
+          const stringifiedChainId = String(token.chainId)
+
+          if (stringifiedChainId === l1ChainId) {
+            // The address is from an L1 token
+
+            if (typeof acc[address] === 'undefined') {
+              // First time encountering the token through its L1 address
+              acc[address] = {
+                ...token,
+                type: TokenType.ERC20,
+                l2Address: undefined,
+                allowed: true,
+                tokenLists: []
+              }
+            } else {
+              // Token was already added to the map through its L2 token
+              acc[address] = { ...acc[address], address }
+            }
+
+            const tokenLists = acc[address].tokenLists
+
+            if (!tokenLists.includes(tokenList.bridgeTokenListId)) {
+              acc[address].tokenLists.push(tokenList.bridgeTokenListId)
+            }
+          } else if (stringifiedChainId === l2ChainId) {
+            // The token is an L2 token
+
+            if (!token.extensions?.bridgeInfo) {
+              return
+            }
+
+            // @ts-ignore
+            //
+            // TODO: should we upgrade '@uniswap/token-lists'?
```

Comment:
`@uniswap/token-lists` added support for extension objects in https://github.com/Uniswap/token-lists/commit/3af2199eceaaa2e46b47d211e1c80d609805be63, but they haven't released a version that includes it.

We could set the dependency to always follow their `main` branch, but that could break stuff in the future. Could we reach out to them and ask them to tag a new version?